File: ruby/resistor-color-duo/resistor_color_duo.rb
# frozen_string_literal: true

# ResistorColorDue determines numeric value of two band linear resistor
class ResistorColorDuo
  COLORS = %w[black brown red orange yellow green blue violet grey white].freeze

  def self.value(colors)
    colors.first(2).map(&ResistorColorDuo.new).join.to_i
  end

  def to_proc
    proc { |color| COLORS.index(color) }
  end
end

File: javascript/resistor-color-duo/resistor-color-duo.js
const COLORS = ['black', 'brown', 'red', 'orange', 'yellow', 'green', 'blue', 'violet', 'grey', 'white'];
const findColorValue = color => COLORS.indexOf(color)

export const decodedValue = ([colorOne, colorTwo]) => {
  return findColorValue(colorOne) * 10 + findColorValue(colorTwo);
};
